### DISCUSSION TITLE
Bilinear Form to find the inner product of a scalar trial function and a vector field test function?

### DISCUSSION THREAD

**Discussion by TheBEllis** [mfem/mfem#3698] [Q&A] [answered]
Hi MFEM team,
I was wondering if there exists a bilinear form integrator whose operator looks like so,
$(u, \nabla \cdot \vec{v})$.
Where the function $v$ is from the test space of a vector field which is just multiple copies of a scalar field. I don't think this BLF integrator currently exists within MFEM, but I might be looking in the wrong place. Any help is very appreciated:)

**Comment by mlstowell** [comment, marked as answer]
Hi, @TheBEllis,

I believe we have two different bilinear form integrators which are equivalent to this although neither is written in quite that manner. The first is called `GradientIntegrator` and it implements $(\lambda\nabla u, \vec{v})\approx-\left(u, \nabla\cdot(\lambda \vec{v})\right)$. The second is called `VectorDivergenceIntegrator` and it implements $(\lambda\nabla\cdot\vec{u},v)$ which is equivalent to the transpose of $(\lambda u,\nabla\cdot\vec{v})$. One main difference is the location of the optional $\lambda$ coefficient. If you have a non-constant coefficient then you must be careful to pick the correct integrator. If your coefficient is spatially constant either of these will do but the `VectorDivergenceIntegrator` may be the better choice. This is because the `GradientIntegrator` involves an integration-by-parts which gives rise to a boundary integral. Such boundary integrals can be used to enforce Neumann boundary conditions but with a rectangular operator such as this its meaning is less clear.

Best wishes,
Mark

**Reply by TheBEllis** [reply]
Hi Mark,
Thanks for a brilliant run through on the options available, I will take a look and see if I can get something working:)
Kind regards,
Bill
